### PR TITLE
FLUT-839471 - [Others] Add use case code snippets

### PR DESCRIPTION
**Pull request by immankumarsync**
Added the use case code snippets in pdf viewer -> form filling section

### CODE REVIEW

File: Flutter/pdf-viewer/form-filling.md [hunk 1]
```diff
@@ -297,6 +297,119 @@ Widget build(BuildContext context) {
 {% endhighlight %}
 {% endtabs %}
 
+## How do I create and display a custom signature pad?
+
+With the above option available in the `SfPdfViewer`, you can easily hide the built-in signature pad and create and display a custom signature pad to draw and add personalised signatures to the signature form field. The following code example explains the same.
+
+In this example, the custom signature pad using `SfSignaturePad` will be displayed when tapping on the signature field with the following options:
+
+* **Clear** - To clear all the signature strokes in the `SfSignaturePad`.
```
Comment (ramkumarsync) [inline Flutter/pdf-viewer/form-filling.md:306]: Clears all the signature strokes in the `SfSignaturePad`

File: Flutter/pdf-viewer/form-filling.md [hunk 2]
```diff
@@ -609,3 +722,326 @@ Widget build(BuildContext context) {
...
+
+In this example, we are validating the form field data in a registration form when saving. The validation includes,
+<table>
+<tr>
+<td>
+Name
+</td>
+<td>
+* Name is required.<br>
+* Name should be atleast 3 characters.<br>
```
Comment (ramkumarsync) [inline Flutter/pdf-viewer/form-filling.md:872]: Name should have at least 3 characters

File: Flutter/pdf-viewer/form-filling.md [hunk 3]
```diff
@@ -609,3 +722,326 @@ Widget build(BuildContext context) {
...
+
+{% endhighlight %}
+{% endtabs %}
+
+### How do I perform validation over the form field data?
+
+With the help of the `onFormFieldFocusChange` and ` onFormFieldValueChanged ` callbacks and the form field programmatic options available in `SfPdfViewer`, we can validate the form field data. The following code examples explain the same.
+
+In this example, we are validating the form field data in a registration form when saving. The validation includes,
+<table>
```
Comment (ramkumarsync) [inline Flutter/pdf-viewer/form-filling.md:865]: Add headers "Field name" and "Validation messsages"

File: Flutter/pdf-viewer/gesture-callbacks.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ The [onTap](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pd
 
 * `pageNumber` – This property returns the page number on which the tap took place. The value ranges from 1 to the total number of pages in the PDF document. If the tap occurs outside of any PDF page boundaries, the result will be -1.
 
-* `pagePositon` – The property returns the page’s tapped position in the PDF coordinates. The coordinates have their origin at the top-left of the page. The number of the tapped page is identified by the PageNumber property. If the tap occurs outside of any PDF page boundaries, the result will be (-1, -1).
+* `pagePosition` – The property returns the page’s tapped position in the PDF coordinates. The coordinates have their origin at the top-left of the page. The number of the tapped page is identified by the PageNumber property. If the tap occurs outside of any PDF page boundaries, the result will be (-1, -1).
```
Comment (ramkumarsync) [inline Flutter/pdf-viewer/gesture-callbacks.md:20]: Add API reference links

File: Flutter/pdf-viewer/form-filling.md [hunk 5]
```diff
@@ -280,7 +280,7 @@ By default, the `SfPdfViewer` displays the signature pad when tapped on the sign
 
 {% tabs %}
 {% highlight dart hl_lines="9" %}
- 
+
```
Comment (ramkumarsync) [inline Flutter/pdf-viewer/form-filling.md:283]: Remove this unwanted changes

File: Flutter/pdf-viewer/form-filling.md [hunk 6]
```diff
@@ -609,3 +722,326 @@ Widget build(BuildContext context) {
...
+        ],
+      ),
+    );
+  }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### How do I perform validation over the form field data?
```
Comment (ramkumarsync) [inline Flutter/pdf-viewer/form-filling.md:860]: --## How to

File: Flutter/pdf-viewer/overview.md [hunk 7]
```diff
@@ -20,6 +20,8 @@ The Syncfusion [Flutter PDF Viewer](https://www.syncfusion.com/flutter-widgets/f
 * **Page navigation** - Navigate to the desired pages instantly.
 * **Bookmark navigation** - Bookmarks saved in the document are loaded and made ready for easy navigation. This feature helps navigate the topics bookmarked already within the PDF document.
 * **Document link annotation navigation** - Navigate to the desired topic or position by tapping the document link annotation of the topics in the table of contents in a PDF document.
+* **Hyperlink navigation** - Detects hyperlinks and tapping on the hyperlink will open the URL in a browser.
+* **Form filling** - Add and modify the existing form field content in a PDF document.
```
Comment (ramkumarsync) [inline Flutter/pdf-viewer/overview.md:24]: Fill, edit, flatten, save, export, and import AcroForm field data in a PDF document.

File: Flutter/pdf-viewer/overview.md [hunk 8]
```diff
@@ -20,6 +20,8 @@ The Syncfusion [Flutter PDF Viewer](https://www.syncfusion.com/flutter-widgets/f
 * **Page navigation** - Navigate to the desired pages instantly.
 * **Bookmark navigation** - Bookmarks saved in the document are loaded and made ready for easy navigation. This feature helps navigate the topics bookmarked already within the PDF document.
 * **Document link annotation navigation** - Navigate to the desired topic or position by tapping the document link annotation of the topics in the table of contents in a PDF document.
+* **Hyperlink navigation** - Detects hyperlinks and tapping on the hyperlink will open the URL in a browser.
```
Comment (ramkumarsync) [inline Flutter/pdf-viewer/overview.md:23]: Detects hyperlinks, and tapping on the hyperlink will open the URL in a default web browser.

**Review by ramkumarsync — APPROVED** [review]

File: Flutter/pdf-viewer/gesture-callbacks.md [hunk 9]
```diff
@@ -15,11 +15,11 @@ The [SfPdfViewer](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/lat
 
 The [onTap](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/SfPdfViewer/onTap.html) callback triggers when the user taps on the `SfPdfViewer` widget and the [PdfGestureDetails](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/PdfGestureDetails-class.html) returns the following information,
 
-* `pageNumber` – This property returns the page number on which the tap took place. The value ranges from 1 to the total number of pages in the PDF document. If the tap occurs outside of any PDF page boundaries, the result will be -1.
+* [pageNumber](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/PdfGestureDetails/pageNumber.html) – This property returns the page number on which the tap took place. The value ranges from 1 to the total number of pages in the PDF document. If the tap occurs outside of any PDF page boundaries, the result will be -1.
```
Comment (florenceliliana) [inline Flutter/pdf-viewer/gesture-callbacks.md:18]: * [pageNumber](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/PdfGestureDetails/pageNumber.html) – This property returns the page number on which the tap took place. The value ranges from 1 to the total number of pages in the PDF document. If the tap occurs outside any PDF page boundaries, the result will be -1.

File: Flutter/pdf-viewer/gesture-callbacks.md [hunk 10]
```diff
@@ -15,11 +15,11 @@ The [SfPdfViewer](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/lat
...
 
-* `pagePositon` – The property returns the page’s tapped position in the PDF coordinates. The coordinates have their origin at the top-left of the page. The number of the tapped page is identified by the PageNumber property. If the tap occurs outside of any PDF page boundaries, the result will be (-1, -1).
+* [pagePosition](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/PdfGestureDetails/pagePosition.html) – The property returns the page’s tapped position in the PDF coordinates. The coordinates have their origin at the top-left of the page. The number of the tapped page is identified by the PageNumber property. If the tap occurs outside of any PDF page boundaries, the result will be (-1, -1).
```
Comment (florenceliliana) [inline Flutter/pdf-viewer/gesture-callbacks.md:20]: * [pagePosition](https://pub.dev/documentation/syncfusion_flutter_pdfviewer/latest/pdfviewer/PdfGestureDetails/pagePosition.html) – The property returns the page’s tapped position in the PDF coordinates. The coordinates have their origin at the top-left of the page. The number of the tapped page is identified by the PageNumber property. If the tap occurs outside any PDF page boundaries, the result will be (-1, -1).

**Review by florenceliliana — COMMENTED** [review]
I have completed the content review.

**Pull request merged**